### PR TITLE
Updates + HTTP Client Sanity

### PR DESCRIPTION
**Pull request by AffectedArc07**
This PR imports some more updates from TG, as well as new methods for the HTTP client to manually create and close the thread pools.

This is quite possibly, the worst code I have ever written. But it does work.

Closing an already closed pool does not break
Opening an already open pool does not break
Making web requests on a closed pool does not break (It just does nothing)

Logging still works
image

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
Comment (spookydonut) [inline .vscode/launch.json]: you cant use comments in a json

File: src/dmi.rs [hunk 2]
```diff
@@ -57,8 +58,8 @@ fn create_png(path: &str, width: &str, height: &str, data: &str) -> Result<()> {
...
 
     let mut encoder = Encoder::new(File::create(path)?, width, height);
-    encoder.set(png::ColorType::RGB);
-    encoder.set(png::BitDepth::Eight);
+    encoder.set_color(png::ColorType::RGB);
+    encoder.set_depth(png::BitDepth::Eight);
     let mut writer = encoder.write_header()?;
     Ok(writer.write_image_data(&result)?)
-}
+}
```
Comment (spookydonut) [inline src/dmi.rs]: newline

File: src/http.rs [hunk 3]
```diff
@@ -137,18 +158,35 @@ fn submit_request(prep: RequestPrep) -> Result<String> {
...
+    });
+    Some("")
+} }
+
+byond_fn! { shutdown_http_client() {
+    HTTP_CLIENT.with(|cell| {
+        cell.replace(None)
+    });
+    Some("")
+} }
```
Comment (spookydonut) [inline src/http.rs]: newline

File: src/sql.rs [hunk 4]
```diff
@@ -283,4 +285,4 @@ fn params_from_json(params: &str) -> Params {
         Ok(serde_json::Value::Array(a)) => array_to_params(a),
         _ => Params::Empty,
     }
-}
+}
```
Comment (spookydonut) [inline src/sql.rs]: newline

File: src/url.rs [hunk 5]
```diff
@@ -10,17 +11,31 @@ byond_fn! { url_decode(data) {
...
+                    *byte = b' ';
+                }
+            }
+            Cow::Owned(replaced)
+        }
     }
-
-    Ok(decoded)
-}
+}
```
Comment (spookydonut) [inline src/url.rs]: newline

**Review by spookydonut — CHANGES_REQUESTED** [review]
fix your editor

**Comment by AffectedArc07** [comment]
Force pushed to uncommit my `launch.json`

**Review by Absolucy — APPROVED** [review]
lgtm

File: Cargo.toml [hunk 6]
```diff
@@ -1,47 +1,46 @@
...
 serde_json = { version = "1.0", optional = true }
 serde_derive = { version = "1.0", optional = true }
-lazy_static = { version = "1.3", optional = true }
-
-[dependencies.mysql]
-version = "18.2"
-optional = true
-git = "https://github.com/blackbeam/rust-mysql-simple"
-rev = "b039fe7a11609aeab9071289bcff496b9366389a"
+lazy_static = { version = "1.4", optional = true }
```
Comment (Putnam3145) [inline Cargo.toml]: probably don't need this anymore since you're not using it from what i can see

Comment (AffectedArc07) [reply Cargo.toml]: Resolved

**Review by Nukeuler123 — APPROVED** [review]
All of this looks good, even though my IDE is complaining about some error everything compiles just fine.

**Pull request merged**